### PR TITLE
Change error generated files to mmu format

### PR DESCRIPTION
**Pull request by michalrudolf**
BFW-4776

Add
 - std::array<ButtonOperations, 4> buttons; (default ButtonOperations::NoOperation)
 - ErrType type; (default ErrType::ERROR)

to all generated error_codes files. We will use buttons and types on printers without mmu as well in the future.

### CODE REVIEW

File: generate_buddy_headers.py [hunk 1]
```diff
@@ -132,33 +157,34 @@ def generate_header_file(yaml_file_name, header_file_name, mmu, list, includes):
...
-                if len(btns) == 2:
-                    btns.append("ButtonOperations::NoOperation")
+            # Make sure the array always has 3 buttons, NoOperation button
+            # occupying the appropriate "empty" slots
+            if len(btns) == 0:
+                btns.append("ButtonOperations::NoOperation")
+            if len(btns) == 1:
+                btns.insert(0, "ButtonOperations::NoOperation")
+            if len(btns) == 2:
+                btns.append("ButtonOperations::NoOperation")
```
Comment (lukash) [inline generate_buddy_headers.py:171]: This three button "layout" is specific to MMU, I think generic dialogs in the printer can have up to 4 buttons, and they are not spaced this way, they are just placed at the beginning of an array of length 4 and the rest is filled up with some `none`, i.e. `ButtonOperations::NoOperation` in this context.

Comment (michalrudolf) [reply generate_buddy_headers.py]: force-pushed

File: generate_buddy_headers.py [hunk 2]
```diff
@@ -22,6 +22,7 @@
 """#pragma once
 #include "inttypes.h"
 #include "button_operations.h"
+#include "error_types.h"
```
Comment (lukash) [inline generate_buddy_headers.py:25]: Nitpick: I'd name the file `"error_type.h"`

Comment (michalrudolf) [reply generate_buddy_headers.py]: force-pushed

**Review by DRracer — APPROVED** [review]

**Pull request merged**